File: LeetCode/cpp/1460.cpp
//Link: https://leetcode.com/problems/make-two-arrays-equal-by-reversing-sub-arrays/

class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        int count_t[1001] = {0};
        int count_a[1001] = {0};
        for(int i = 0 ; i < target.size() ; i++) {
            count_t[target[i]]++;
            count_a[arr[i]]++;
        }
        for(int i = 0 ; i < 1001 ; i++)
            if(count_t[i] != count_a[i])
                return false;
        return true;
    }
};

File: LeetCode/py/1460.py
#Link: https://leetcode.com/problems/make-two-arrays-equal-by-reversing-sub-arrays/

class Solution:
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        count_t = [0]*1001
        count_a = [0]*1001
        for i in range(len(target)):
            count_t[target[i]] += 1
            count_a[arr[i]] += 1
        for i in range(1001):
            if count_t[i] != count_a[i]:
                return False
        return True
